posthog/hogql/database/database.py

```diff
@@ -505,6 +506,12 @@ def create_hogql_database(
             if database.events.fields.get(mapping.group_type) is None:
                 database.events.fields[mapping.group_type] = FieldTraverser(chain=[f"group_{mapping.group_type_index}"])
 
+    with timings.measure("can_skip_dw_tables"):
+        if skip_dw_tables:
+            # Skip data warehouse tables loading
+            # This is used internally for QueryRunners that know ahead of time that they will not be using the data warehouse tables
+            return database
```

Comment:
Would be interesting if we could instead extract all of the DWH fields to a  `_use_datawarehouse_tables()` function and then make it `if not skip_dw_tables` (or maybe inver the param name `if include_dw_tables`). That might be tricky because it uses a lot of fields from outside. I might take a stab at simplifying this code once you merge this.